src/ResourceManager/Compute.ManagedService/Commands.Compute.ManagedService/Tools/Move-AzureRmVhdVM.ps1

```diff
@@ -0,0 +1,237 @@
+        {
+            $rgName = ($HyperVVMName + (date).Ticks).Substring(0, 21);
+        }
+        New-AzureRmVhdVM -ResourceGroupName $rgName -Location $Location -OSType $OSType -DiskFile $diskFiles;
+    }
+    else
+    {
+        Write-Warning "Cannot find the VHD files for VM '$HyperVVMName'. Exit.";
+    }
+}
+
+function Get-VMOperatingSystemName($computerNameInput, $hyperVVMNameInput)
+{
+    # Filter for parsing XML data
+    filter Import-CimXml
+    {
+        # Create new XML object from input
+        $CimXml = [Xml]$_;
+        $CimObj = New-Object -TypeName System.Object;
+
+        # Iterate over the data and pull out just the value name and data for each entry
+        foreach ($CimProperty in $CimXml.SelectNodes("/INSTANCE/PROPERTY[@NAME='Name']"))
+        {
+            $CimObj | Add-Member -MemberType NoteProperty -Name $CimProperty.NAME -Value $CimProperty.VALUE;
+        }
+
+        foreach ($CimProperty in $CimXml.SelectNodes("/INSTANCE/PROPERTY[@NAME='Data']")) 
+        {
+            $CimObj | Add-Member -MemberType NoteProperty -Name $CimProperty.NAME -Value $CimProperty.VALUE;
+        }
+        # Display output
+        $CimObj;
+    }
+
+    # Get the virtual machine object
+    $query = "Select * From Msvm_ComputerSystem Where ElementName='" + $hyperVVMNameInput + "'";
+    $Vm = gwmi -Namespace root\virtualization\v2 -Query $query -ComputerName $computerNameInput;
+
+    # Get the KVP Object
+    $query = "Associators of {$Vm} Where AssocClass=Msvm_SystemDevice ResultClass=Msvm_KvpExchangeComponent";
+    $Kvp = gwmi -Namespace root\virtualization\v2 -Query $query -ComputerName $computerNameInput;
+
+    #Write-Host "Guest KVP information for: '$hyperVVMNameInput'";
+
+    # Filter the results
+    try
+    {
+        $osNameItem = $Kvp.GuestIntrinsicExchangeItems | Import-CimXml | where Name -eq "OSName";
+        Write-Output $osNameItem.Data;
+    }
+    catch
+    {
+        Write-Verbose "OS type name not found...";
+    }
+}
+
+function Export-VMWithProgress($computerName, $vmName, $exportDir)
+{
+    # Record the supposed to be exported VHD files first
+    $vmObj = Get-VM -ComputerName $computerName -Name $vmName;
+    if (-not $vmObj.HardDrives -or $vmObj.HardDrives.Count -le 0)
+    {
+        throw "No hard drives can be found from VM '$vmName'...";
+    }
+    [string[]]$vmDiskFileNames = @();
+    foreach ($hd in $vmObj.HardDrives)
+    {
+        $vmDiskFileNames += (Split-Path -Path $hd.Path -Leaf);
+    }
+    
+    # Start the real export process
+    $exportJob = Export-VM -ComputerName $computerName -Name $vmName -Path $exportDir -AsJob;
+    while( $exportJob.State -eq "Running" -or $exportJob.State -eq "NotStarted") 
+    {
+        Write-Host ("[Export] " + $($exportJob.Progress.PercentComplete) + "% completed...");
+        sleep(5);
+    }
+
+    if($exportJob.State -ne "Completed")
+    {
+        Write-Error ("Export-VM did not complete: " + $exportJob.State);
```

Comment:
throwing will write the exception into the error stream.  If this is intended as progress, perhaps it belongs int he Progress stream?

Reply:
removed